numpy/core/_add_newdocs.py

```diff
@@ -1623,10 +1632,10 @@
         between two adjacent values, ``out[i+1] - out[i]``.  The default
         step size is 1.  If `step` is specified as a position argument,
         `start` must also be given.
-    dtype : dtype
+    dtype : dtype, optional
         The type of the output array.  If `dtype` is not given, infer the data
         type from the other input arguments.
-    ${ARRAY_FUNCTION_LIKE}
+    ${ARRAY_FUNCTION_LIKE} This is an optional argument.
```

Comment:
I don't know if there are better ways of doing this - I think marking the original `array_like` as optional will possibly create problems for other functions.

Reply:
Seems better to include in the template, it is always optional.